app/pluginmanager.js

```diff
@@ -1151,12 +1153,17 @@ PluginManager.prototype.checkPluginDependencies = function (folder) {
         if(package_json.engines.node) {
             result.nodeCheck = true;
             var nodeVersion = semver.coerce(process.versions.node);
+            var pluginNodeVersion = semver.coerce(package_json.engines.node);
             if(nodeVersion === null) {
                 result.success = 'warn';
                 result.message += 'Current Node version cannot be detected. ';
             } else if(!semver.satisfies(nodeVersion, package_json.engines.node)) {
                 result.success = 'failed';
                 result.message += 'Node version ' + nodeVersion + ' not usable with plugin. ';
+            } else if(native_modules.length > 0 && nodeVersion.major !== pluginNodeVersion.major) {
```

Comment:
@ashthespy this implies that we add the property native modules to the plugin info?
If yes, how?

Reply:
No, `native_modules` is determined a few lines above ;)